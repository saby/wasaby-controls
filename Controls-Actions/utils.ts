import { IContextValue } from 'Controls/context';
import { Record } from 'Types/entity';

export interface IContextData extends IContextValue {
    documentContext?: Record;
}

/**
 * Получает объект из контекста
 * @public
 * @param context
 * @param objectName
 */
function findObject(context: IContextData, objectName: string): unknown {
    return context?.[objectName];
}

export { findObject };
