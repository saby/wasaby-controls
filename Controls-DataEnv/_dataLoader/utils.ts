import { loadAsync } from 'WasabyLoader/ModulesLoader';
import { addPageDeps } from 'UI/Deps';
import { logger as Logger } from 'Application/Env';
import { RPC } from 'Browser/Transport';

function resolveLogger(error: Error): typeof Logger.error | typeof Logger.warn {
    let logger = Logger.error;

    if (error instanceof RPC.Error) {
        logger = error.errType === 'warning' ? Logger.warn : Logger.error;
    }

    return logger;
}

export async function loadModule<TModule>(moduleName: string): Promise<TModule | undefined> {
    try {
        const module = await loadAsync<TModule>(moduleName);
        addPageDeps([moduleName]);
        return module;
    } catch (error: unknown) {
        loadErrorHandler(
            error,
            `Controls-DataEnv/dataLoader::Ошибка при загрузке модуля ${moduleName}`
        );
    }
}

export function loadErrorHandler(e: unknown, message: string): void {
    if (e instanceof Error) {
        const logger = resolveLogger(e);
        logger(`${message} ${e.message}`);
    }
}
