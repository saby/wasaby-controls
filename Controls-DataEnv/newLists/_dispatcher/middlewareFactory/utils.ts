export const addName = <T extends Function>(label: string, fn: T): T => {
    const wrapper = {
        [label]: ((...args: unknown[]) => fn(...args)) as unknown as T,
    };

    return wrapper[label];
};

const POSTFIX = '{name}';
export const createMask = <T extends string>(prefix: T) => `${prefix}(${POSTFIX})` as const;
export const decorateName = (mask: string, name: string) => mask.replace(POSTFIX, name);
export const undecorateName = (mask: string, decoratedName: string) => {
    const [start, end] = mask.split(POSTFIX);
    return decoratedName.replace(start, '').replace(end, '');
};
