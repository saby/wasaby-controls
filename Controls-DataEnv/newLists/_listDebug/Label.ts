/**
 * @kaizen_zone 039c82f1-a0a3-4548-82d6-c9e1dbaf5de0
 */
import type { Slice } from 'Controls-DataEnv/slice';

const SYNTHETIC_NAME = '_dataSyntheticStoreId';

let LABELS_COUNT: number;
let LABELS: WeakMap<
    Slice,
    {
        name: string;
        isSynthetic: boolean;
    }
>;
(() => {
    LABELS_COUNT = 0;
    LABELS = new WeakMap();
})();

export function initLabel(instance: Slice, name: string): string {
    let resultName: string;

    if (name === SYNTHETIC_NAME) {
        LABELS_COUNT++;
        resultName = `SYNTHETIC #${LABELS_COUNT}`;
        LABELS.set(instance, {
            name: resultName,
            isSynthetic: true,
        });
    } else {
        resultName = name;
        LABELS.set(instance, { name, isSynthetic: false });
    }

    return resultName;
}

export function deleteLabel(instance: Slice): void {
    LABELS.delete(instance);
}

export function getLabel(instance: Slice): string {
    const info = LABELS.get(instance);
    if (info) {
        return info.name;
    }

    return '';
}
