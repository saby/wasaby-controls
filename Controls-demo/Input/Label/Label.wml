<div class="controlsDemo__wrapper  controlsDemo__flex controlsDemo__flex-flow_wrap controlsDemo_widthFit">
    <div class="controlsDemo__cell controls-margin_right-l">
        <Router.router:Reference attr:class="controls-text-label"
                                 state="app/:app"
                                 app="{{ 'Controls-demo/Input/Label/Base/Index' }}">
            <a href="{{content.href}}">Базовая настройка</a>
        </Router.router:Reference>
        <Controls-demo.Input.Label.Base.Index/>
    </div>
    <div class="controlsDemo__cell controls-margin_right-l">
        <Router.router:Reference attr:class="controls-text-label"
                                 state="app/:app"
                                 app="{{ 'Controls-demo/Input/Label/FontColorStyle/Index' }}">
            <a href="{{content.href}}">Настройка fontColorStyle</a>
        </Router.router:Reference>
        <Controls-demo.Input.Label.FontColorStyle.Index/>
    </div>
    <div class="controlsDemo__cell controls-margin_right-l">
        <Router.router:Reference attr:class="controls-text-label"
                                 state="app/:app"
                                 app="{{ 'Controls-demo/Input/Label/FontSize/Index' }}">
            <a href="{{content.href}}">Настройка fontSize</a>
        </Router.router:Reference>
        <Controls-demo.Input.Label.FontSize.Index/>
    </div>
    <div class="controlsDemo__cell controls-margin_right-l">
        <Router.router:Reference attr:class="controls-text-label"
                                 state="app/:app"
                                 app="{{ 'Controls-demo/Input/Label/Href/Index' }}">
            <a href="{{content.href}}">Настройка href</a>
        </Router.router:Reference>
        <Controls-demo.Input.Label.Href.Index/>
    </div>
    <div class="controlsDemo__cell controls-margin_right-l">
        <Router.router:Reference attr:class="controls-text-label"
                                 state="app/:app"
                                 app="{{ 'Controls-demo/Input/Label/Underline/Index' }}">
            <a href="{{content.href}}">Настройка underline</a>
        </Router.router:Reference>
        <Controls-demo.Input.Label.Underline.Index/>
    </div>
</div>
