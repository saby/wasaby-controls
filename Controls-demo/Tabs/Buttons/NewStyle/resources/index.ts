import { RecordSet } from 'Types/collection';

export const ITEMS = new RecordSet({
    keyProperty: 'id',
    rawData: [
        {
            id: '1',
            title: 'Document',
        },
        {
            id: '2',
            title: 'Files',
        },
        {
            id: '3',
            title: 'Orders',
        },
    ],
});

export const ITEMS_WITH_ICON_AND_COUNTER = new RecordSet({
    keyProperty: 'id',
    rawData: [
        {
            id: '1',
            title: 'Document',
        },
        {
            id: '2',
            title: 'Files',
            mainCounter: 3,
        },
        {
            id: '3',
            title: 'Orders',
            icon: 'icon-EmptyMessage',
            iconStyle: 'label',
        },
    ],
});
