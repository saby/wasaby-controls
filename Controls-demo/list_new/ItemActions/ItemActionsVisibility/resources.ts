import { IoC } from 'Env/Env';
import { Model } from 'Types/entity';

import { IItemAction, TItemActionShowType } from 'Controls/itemActions';

export interface ISrcData {
    key: number;
    title: string;
    description: string;
}

export const columns = [
    {
        displayProperty: 'key',
    },
    {
        displayProperty: 'title',
    },
    {
        displayProperty: 'description',
    },
];

export const srcData: ISrcData[] = [
    {
        key: 1,
        title: 'Notebooks 1',
        description: 'Другое название 1',
    },
    {
        key: 2,
        title: 'Notebooks 2',
        description: 'Описание вот такое',
    },
    {
        key: 3,
        title: 'Smartphones 3 ',
        description: 'Хватит страдать',
    },
    {
        key: 4,
        title: 'Notebooks 1',
        description: 'Другое название 1',
    },
    {
        key: 5,
        title: 'Notebooks 2',
        description: 'Описание вот такое',
    },
    {
        key: 6,
        title: 'Smartphones 3 ',
        description: 'Хватит страдать',
    },
];

export const menuItemActions: IItemAction[] = [
    {
        id: 1,
        icon: 'icon-PhoneNull',
        title: 'phone',
        handler(model: Model): void {
            IoC.resolve('ILogger').info('action phone Click ', model);
        },
    },
    {
        id: 2,
        icon: 'icon-EmptyMessage',
        title: 'message',
        parent: null,
        'parent@': true,
        handler(model: Model): void {
            alert('Message Click');
        },
    },
    {
        id: 3,
        icon: 'icon-Profile',
        title: 'profile',
        showType: TItemActionShowType.MENU,
        parent: 2,
        'parent@': null,
        handler(model: Model): void {
            IoC.resolve('ILogger').info('action profile Click');
        },
    },
    {
        id: 6,
        title: 'call',
        parent: 2,
        'parent@': null,
        handler(model: Model): void {
            IoC.resolve('ILogger').info('action profile Click');
        },
    },
];

export const menuToolbarItemActions: IItemAction[] = [
    {
        id: 5,
        title: 'прочитано',
        showType: TItemActionShowType.TOOLBAR,
        handler(model: Model): void {
            IoC.resolve('ILogger').info('action read Click');
        },
    },
    ...menuItemActions,
    {
        id: 4,
        icon: 'icon-Erase',
        iconStyle: 'danger',
        title: 'delete pls',
        showType: TItemActionShowType.TOOLBAR,
        handler(model: Model): void {
            IoC.resolve('ILogger').info('action delete Click');
        },
    },
];
