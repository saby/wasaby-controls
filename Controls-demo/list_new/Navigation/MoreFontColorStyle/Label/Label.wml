<div class="controlsDemo__cell controlsDemo_fixedWidth500 moreFontColorStyle">
    <Controls.scroll:Container>
        <Controls.list:View
                storeId="NavigationMoreFontColorStyleLabel"
                moreFontColorStyle="label"/>
    </Controls.scroll:Container>
</div>
