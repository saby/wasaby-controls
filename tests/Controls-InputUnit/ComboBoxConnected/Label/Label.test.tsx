/**
 * @jest-environment jsdom
 */

import { getContent } from 'Controls-InputUnit/resources/utils';
import { WasabyEvents } from 'UICore/Events';
import { unmountComponentAtNode } from 'react-dom';
import { render } from '@testing-library/react';
import { default as Label } from 'Controls-Input-demo/ComboboxConnected/Label';

describe('Controls-Input/ComboBoxGroupConnected/Label', () => {
    setTestID(152174957);
    let container = null;

    beforeEach(() => {
        jest.useFakeTimers('modern');
        jest.setSystemTime(new Date(1996, 5, 21));
        container = document.createElement('div');
        WasabyEvents.initInstance(container);
        document.body.appendChild(container);
    });
    afterEach(() => {
        unmountComponentAtNode(container);
        WasabyEvents.destroyInstance(container);
        jest.useRealTimers();
        container.remove();
        container = null;
    });

    it('Label', async () => {
        render(getContent(Label), { container });
        expect(container).toMatchSnapshot();
    });
});
