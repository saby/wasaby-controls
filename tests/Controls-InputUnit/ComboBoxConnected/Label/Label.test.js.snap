// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls-Input/ComboBoxGroupConnected/Label Label 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo_fixedWidth300 tw-flex tw-flex-col"
  >
    <div
      class="controls-ComboBox controls_toggle_theme-undefined controls-ComboBox-borderStyle-valid   controls-ComboBox-borderStyle-valid_focusing controls-ComboBox_background-same controls-text-default controls-fontweight-default controls-inlineheight-m controls-Combobox-fontsize-m controls-ComboBox_textAlign-left controls-ComboBox_border-partial controls-margin_top-m controls-margin_top-m  tw-w-full"
      tabindex="0"
      title=""
    >
      <div
        class="controls-ComboBox-wrapper"
      >
        <div
          class="controls-ComboBox__field controls-ComboBox__field_margin-null"
        >
          <div
            class="controls-ComboBox__placeholder controls-ComboBox__contentTemplate  controls-ComboBox__placeholderColor tw-cursor-text"
            tabindex="0"
          >
            Выберите вариант
          </div>
        </div>
        <span
          class="controls-ComboBox__arrow-wrapper controls-ComboBox__arrow-wrapper_offset-null"
        >
          <span
            class="controls-ComboBox__iconArrow icon-ExpandLight"
          />
        </span>
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-col controls-margin_top-m "
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0 controls-Label_verticalPosition"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            label top
          </div>
          
        </div>
      </a>
      <div
        class="controls-ComboBox controls_toggle_theme-undefined controls-ComboBox-borderStyle-valid   controls-ComboBox-borderStyle-valid_focusing controls-ComboBox_background-same controls-text-default controls-fontweight-default controls-inlineheight-m controls-Combobox-fontsize-m controls-ComboBox_textAlign-left controls-ComboBox_border-partial controls-margin_top-m"
        tabindex="0"
        title=""
      >
        <div
          class="controls-ComboBox-wrapper"
        >
          <div
            class="controls-ComboBox__field controls-ComboBox__field_margin-null"
          >
            <div
              class="controls-ComboBox__placeholder controls-ComboBox__contentTemplate  controls-ComboBox__placeholderColor tw-cursor-text"
              tabindex="0"
            >
              Выберите вариант
            </div>
          </div>
          <span
            class="controls-ComboBox__arrow-wrapper controls-ComboBox__arrow-wrapper_offset-null"
          >
            <span
              class="controls-ComboBox__iconArrow icon-ExpandLight"
            />
          </span>
        </div>
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-row tw-items-baseline controls-margin_top-m "
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0 tw-flex-shrink"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-m  controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            label start
          </div>
          
        </div>
      </a>
      <div
        class="controls-ComboBox controls_toggle_theme-undefined controls-ComboBox-borderStyle-valid   controls-ComboBox-borderStyle-valid_focusing controls-ComboBox_background-same controls-text-default controls-fontweight-default controls-inlineheight-m controls-Combobox-fontsize-m controls-ComboBox_textAlign-left controls-ComboBox_border-partial controls-input_label_min-input-content controls-margin_top-m"
        tabindex="0"
        title=""
      >
        <div
          class="controls-ComboBox-wrapper"
        >
          <div
            class="controls-ComboBox__field controls-ComboBox__field_margin-null"
          >
            <div
              class="controls-ComboBox__placeholder controls-ComboBox__contentTemplate  controls-ComboBox__placeholderColor tw-cursor-text"
              tabindex="0"
            >
              Выберите вариант
            </div>
          </div>
          <span
            class="controls-ComboBox__arrow-wrapper controls-ComboBox__arrow-wrapper_offset-null"
          >
            <span
              class="controls-ComboBox__iconArrow icon-ExpandLight"
            />
          </span>
        </div>
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-row tw-items-baseline controls-margin_top-m "
    >
      <span
        class="controls-icon_size-s controls-icon_style-secondary controls-margin_right-s icon-SabyBird controls-icon"
      />
      <div
        class="controls-ComboBox controls_toggle_theme-undefined controls-ComboBox-borderStyle-valid   controls-ComboBox-borderStyle-valid_focusing controls-ComboBox_background-same controls-text-default controls-fontweight-default controls-inlineheight-m controls-Combobox-fontsize-m controls-ComboBox_textAlign-left controls-ComboBox_border-partial controls-margin_top-m"
        tabindex="0"
        title=""
      >
        <div
          class="controls-ComboBox-wrapper"
        >
          <div
            class="controls-ComboBox__field controls-ComboBox__field_margin-null"
          >
            <div
              class="controls-ComboBox__placeholder controls-ComboBox__contentTemplate  controls-ComboBox__placeholderColor tw-cursor-text"
              tabindex="0"
            >
              Выберите вариант
            </div>
          </div>
          <span
            class="controls-ComboBox__arrow-wrapper controls-ComboBox__arrow-wrapper_offset-null"
          >
            <span
              class="controls-ComboBox__iconArrow icon-ExpandLight"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
